src/Bicep.Core/Semantics/ParamsSemanticModel.cs

```diff
@@ -39,86 +42,67 @@ public ParamsSemanticModel(BicepParamFile bicepParamFile, Compilation? bicepComp
         /// <returns></returns>
         private IEnumerable<IDiagnostic> GetAdditionalSemanticDiagnostics()
         {
+            var diagnosticWriter = ToListDiagnosticWriter.Create();
```

Comment:
`diagnosticWriter` is not used if `this.BicepCompilation` is null. Let's put it after the if statement.